src/components/molecules/CreateVenueModal/CreateVenueModal.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useCallback, useState } from "react";
+import { Form, Modal } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+
+import { createUrlSafeName, createVenue_v2 } from "api/admin";
+
+import { useUser } from "hooks/useUser";
+import { useVenueId } from "hooks/useVenueId";
+
+import { venueLandingUrl } from "utils/url";
+
+import { VenueTemplate } from "types/venues";
+
+import { newVenueSchema } from "pages/Admin/Details/ValidationSchema";
+
+import "./CreateVenueModal.scss";
+
+interface CreateVenueModalProps {
+  isVisible: boolean;
+  onHide?: () => void;
+}
+
+export interface FormValues {
+  name: string;
+}
+
+export const CreateVenueModal: React.FC<CreateVenueModalProps> = ({
+  isVisible,
+  onHide,
+}) => {
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const venueId = useVenueId();
+  const { user } = useUser();
+
+  const hide = useCallback(() => {
+    onHide && onHide();
+  }, [onHide]);
+
+  const onSubmit = useCallback(
+    async (vals: FormValues) => {
+      if (!user || isLoading) return;
+      setLoading(true);
+
+      try {
+        const venue = {
+          name: vals.name,
+          template: VenueTemplate.partymap,
+        };
+        await createVenue_v2(venue, user);
+        hide();
+      } catch (e) {
+        console.error(e);
+      } finally {
+        setLoading(false);
+      }
+    },
```

Comment:
Instead of `try`/`catch`/`await` this would probably be simpler just using the chained Promise methods.

```typescript
createVenue_v2(venue, user)
  .then(() => {
    hide()
  })
  .catch((e) => {
    // TODO: don't just console.error, Bugsnag.notify and/or display this to the user in some way
  })
  .finally(() => {
    setLoading(false)
  })
```

This aligns with how we tend to do things elsewhere in components/etc. `async`/`await` becomes far more useful/boilerplate reducing in bigger/more complex situations, but it becomes overkill/more verbose in simple situations like this.

Reply:
Noted.